### ISSUE TITLE
Cannot install on Ubuntu 20.04 or on CentOS 8

### ISSUE THREAD

**Issue by airdogvan** [trailofbits/algo#14486]
I'll just copy what I posted on Gitter a few days ago:

Just for reference this was tried on a brand new install of Ubuntu 20.04

starting a new install, (Ubuntu 20.04) and after the command : python3 -m virtualenv --python="$(command -v python3)" .env && source .env/bin/activate && python3 -m pip install -U pip virtualenv && python3 -m pip install -r requirements.txt I get the error: No module named 'virtualenv.seed.via_app_data'
running from algo dir (git clone)
did many successful install before without problems...

Jun 03 10:55
hmm now becoming problematic on 20.04.
googled and the solution was pip3 uninstall virtualenv
after that the script began to run but only to issue multiple errors later the beginning of which is: File "/root/algo/.env/lib/python3.8/site-packages/pip/_internal/cli/base_command.py", line 167, in exc_logging_wrapper
so seems that algo cannot be installed on 20.04 anymore... or am I wrong?

Jun 03 11:01
this new error came after "Collecting ansible-core==2.12.3"

Tried also on CentOS 8 with no success.  Error is:

ERROR: Could not find a version that satisfies the requirement ansible-core==2.12.3 (from versions: 0.0.1a1, 2.11.0b1, 2.11.0b2, 2.11.0b3, 2.11.0b4, 2.11.0rc1, 2.11.0rc2, 2.11.0, 2.11.1rc1, 2.11.1, 2.11.2rc1, 2.11.2, 2.11.3rc1, 2.11.3, 2.11.4rc1, 2.11.4, 2.11.5rc1, 2.11.5, 2.11.6rc1, 2.11.6, 2.11.7rc1, 2.11.7, 2.11.8rc1, 2.11.8, 2.11.9rc1, 2.11.9, 2.11.10rc1, 2.11.10, 2.11.11rc1, 2.11.11, 2.11.12rc1, 2.11.12)            
ERROR: No matching distribution found for ansible-core==2.12.3

after trying to run script: python3 -m virtualenv --system-site-packages  --python="$(command -v python3)" .env &&   source .env/bin/activate &&   python3 -m pip install -U pip virtualenv &&   python3 -m pip install -r requirements.txt

So could someone tell me which distros Algo can now be installed on?  I don't have access on my provider to Fedora.  The only alternative available for me that I have not tried is Debian.

But not sure what has changed for Ubuntu (and BTW anything that cannot be installed on Ubuntu is not going to have a large following) but did many successful install on 20.04 before without problems.

**Comment by davidemyers** [comment]
I have no problems running Algo on Ubuntu Server 20.04. I haven't tried Ubuntu Desktop recently though, is that what you're using?

**Comment by airdogvan** [comment]
Not sure what you mean, as far as I know only 1 version of 20.04.  On the machine I was trying to install no desktop.  It's a VPS, so no GUI.

**Comment by davidemyers** [comment]
[Ubuntu](https://ubuntu.com/desktop) and [Ubuntu Server](https://ubuntu.com/server) are different and it wasn't clear which one you were using. While you can only turn an instance of Ubuntu Server into an AlgoVPN, you can install and run the Algo scripts on other systems to create an AlgoVPN out of a remote instance of Ubuntu Server, which is what I assumed you were doing since you mentioned also trying CentOS.

You didn't mention you were using a VPS. What provider? It worked for me on DigitalOcean a couple days ago. Make sure all of the packages on your VPS instance are up-to-date before installing Algo as some providers don't keep their installation images current.

**Comment by airdogvan** [comment]
First according to dpkg -l ubuntu-desktop, I'm running the server.

Second reinstalled, updated and upgraded, and after running the script:
python3 -m virtualenv --python="$(command -v python3)" .env &&
  source .env/bin/activate &&
  python3 -m pip install -U pip virtualenv &&
  python3 -m pip install -r requirements.txt

this is the error I get.

  Downloading ansible-core-2.12.3.tar.gz (7.8 MB)
     ---------------------------------------- 7.8/7.8 MB 33.2 MB/s eta 0:00:00
ERROR: Exception:
Traceback (most recent call last):                                                                                                                                                                                  
  File "/root/algo/.env/lib/python3.8/site-packages/pip/_internal/cli/base_command.py", line 167, in exc_logging_wrapper                                        

Just wanted to add to was trying to install to localhost (that's always what I have done and it has always worked no matter the provider).           

Tried the install from my local machine (installed many years ago) to the remote and that worked.

But as said local installed on that same provider 1 year ago and it worked fine (with a few workarounds because ansible and other packages versions different.)

So bottom line have it installed and working now but had to do a remote install from a machine installed years ago.

**Comment by happy201807** [comment]
The latest version of Ubuntu 20.04.4 LTS installation has this problem, described as above, older versions are fine.
